File: admin/src/types/category.ts
/**
 * Category related types
 */

import { BaseEntity, PaginationParams, SearchParams, SortParams } from './common';

export interface BaseCategory {
  name: string;
  description?: string;
  icon?: string;
}

export interface Category extends BaseEntity, BaseCategory {
  order: number;
  is_active: boolean;
  prompts_count: number;
}

export interface CategoryCreate extends BaseCategory {
  order?: number;
}

export interface CategoryUpdate {
  name?: string;
  description?: string;
  icon?: string;
  order?: number;
  is_active?: boolean;
}

export interface CategorySummary {
  id: string;
  name: string;
  icon?: string;
  prompts_count: number;
}

export interface CategoryAdmin extends Category {
  created_by?: string;
}

export interface CategoryFilters extends PaginationParams, SearchParams, SortParams {
  is_active?: boolean;
}

export interface CategoryStats {
  total_categories: number;
  active_categories: number;
}

File: backend/app/schemas/category.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime


# Base schemas  
class CategoryBase(BaseModel):
    """Base category schema - simplified"""
    name: str
    description: Optional[str] = None
    icon: Optional[str] = None


# Request schemas
class CategoryCreate(CategoryBase):
    """Category creation schema"""
    order: int = 0


class CategoryUpdate(BaseModel):
    """Category update schema"""
    name: Optional[str] = None
    description: Optional[str] = None
    icon: Optional[str] = None
    order: Optional[int] = None
    is_active: Optional[bool] = None


# Response schemas
class CategoryResponse(CategoryBase):
    """Category response schema"""
    id: str
    order: int
    is_active: bool
    prompts_count: int
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True


class CategorySummary(BaseModel):
    """Category summary schema for lists"""
    id: str
    name: str
    icon: Optional[str] = None
    prompts_count: int
    
    class Config:
        from_attributes = True


class CategoryStats(BaseModel):
    """Category statistics schema"""
    total_categories: int
    active_categories: int


# Admin schemas
class CategoryAdmin(CategoryResponse):
    """Admin category schema"""
    created_by: Optional[str] = None
